Controller/ProfilerController.php

```diff
@@ -35,6 +35,49 @@ class ProfilerController extends ContainerAware
      *
      * @return Response A Response instance
      */
+    public function executeAction($token, $connectionName, $query)
+    {
+        /** @var $profiler \Symfony\Component\HttpKernel\Profiler\Profiler */
+        $profiler = $this->container->get('profiler');
+        $profiler->disable();
+
+        $profile = $profiler->loadProfile($token);
+        $queries = $profile->getCollector('db')->getQueries();
+
+        if (!isset($queries[$connectionName][$query])) {
+            return new Response('This query does not exist.');
+        }
+
+        $query = $queries[$connectionName][$query];
+        if (!$query['explainable']) {
+            return new Response('This query cannot be explained.');
```

Comment:
Do you think we should remove the if condition or just change message text to something like 'This query cannot be executed.'